### DISCUSSION TITLE
Searching releases feature feels a bit half-baked

### DISCUSSION THREAD

**Discussion by mnquintana** [community/community#6006] [Other Feature Feedback, Questions, & Ideas] [answered]
Let me start by saying I am incredibly grateful for the newly added ability to search releases – this is an absolute GODSEND, and resolves one of my longest standing GitHub papercuts. 🙏🏽 

That said, I'm a bit disappointed that all search seems to do is a basic string match on the title, release notes, and (maybe?) the tag, without even highlighting matches in search results.

## Proposal
- It would be incredibly useful if you could not just string search, but also do a faceted search:
  - I want to be able to explicitly filter by:
    - commit
    - author
    - contributors
    - reactions
    - attached assets (name, file type, size)
    - wildcards
    - release name
    - tag name
    - negations
- I also really want to be able to sort releases, and not just rely on the current chronological order:
  - reverse chronological order
  - A-Z / Z-A (by release name / tag name)
  - recently / least recently updated
  - most reactions
- Highlight matching search terms in search results
  - Currently, if you do a search, it's impossible to tell what fields the search is even matching on
- More or less, I'd love to see searching releases have all the power issues / pull requests search currently have and more:
  ![image](https://user-images.githubusercontent.com/823545/136056003-335640b2-d355-4866-b518-c497c338ad0e.png)

This all might sound like overkill (and maybe it is for small side projects), but for real world projects I think it's essential for managing complex release processes at scale.

Let me know if there's any more information I could add that'd be helpful, and thanks again for all your hard work revamping releases! 🙇🏽

**Comment by MylesBorins** [comment, marked as answer]
@mnquintana we are actively working on improvements to search right now including the ability to search by tag and creation date (including the ability to search for date ranges after and before a certain date.

We are also actively exploring getting highlight to work on search results.

The one thing that is not currently in scope is various sorts, we don't have the bandwidth at this point to prioritize that work.

**Reply by MylesBorins** [reply]
significant search updates landed this week including special search syntax shared below.

We are working on expanding all results by default and highlighting terms in the coming weeks

---

## Search syntax for searching releases in a repository

You can provide text in your search query which will be matched against the title, body, and tag of the repository's releases. You can also combine the following qualifiers to target specific releases.

| Qualifier        | Example
| ------------- | -------------
| `draft:true` | **draft:true** will only match draft releases.
| `draft:false` | **draft:false** will only match published releases.
| <code>tag:<em>TAG</em></code> | **tag:v1** matches a release with the v1 tag and any minor or patch versions within v1, such as v1.0, v1.2, and v1.2.5.
| <code>created:<em>DATE</em></code> | **created:2021** will match releases created from 2021. You can also provide date ranges. For more information, see "[Understanding the search syntax](/search-github/getting-started-with-searching-on-github/understanding-the-search-syntax#query-for-dates)."

**Reply by mnquintana** [reply]
@MylesBorins Oh fantastic! Are there any plans to allow partial matches on tag names in search queries? We namespace all our tags (e.g. `desktop/v3.8.1`), and it looks like with the tag search, there's no way to search for partial matches.

What I'd like to do is be able to run a search query for all our Desktop releases:

```
tag:desktop/
```

Which would return every tag that starts with `desktop/`, but that currently returns nothing.

**Reply by MylesBorins** [reply]
This looks like a bug in how we filter tags, it might have to do with the `/` character because I am able to do partial matches of tags such as 

https://github.com/nodejs/node/releases?q=tag%3Av8&expanded=true

Will dig in with team to see what's going on

**Reply by mnquintana** [reply]
@MylesBorins Ah so the partial matches work if I supply a major version number, e.g.: `tag:desktop/v3`, but not if I supply no version number – e.g. `tag:desktop/` returns no matches.

**Reply by MylesBorins** [reply]
From engineering

> Searching for `tag:v2.` (note the ending period) won't work either. Reason being that our search generates clean tokens without delimiters. For the `tag_name` field, we generate these 3 incremental tokens in your example: `test/1`, `test/1.2`, `test/1.2.3` so we can search for any of these, but not for partials.
>
> Searching for partials would result in `tag:v2` matching also `v20.0.0`, `v21.0.0`...


The next step we have right now is to document this behavior since it doesn't look like there is an easy fix

**Reply by coyoteecd** [reply]
@MylesBorins even with the update, the "search releases" is still lacking some aspects. Here's a good example that I happen to use in one of our projects: https://github.com/dotansimha/graphql-code-generator/releases?q=%22%40graphql-codegen%2Ftypescript%22&expanded=true. This monorepo has a big bunch of packages, and the author uses the releases feature to publish their changelog. It would be great if the quoted search in releases returned **only exact matches**. As it is now, the results are practically unusable, because there are many packages that start with the same search string, and github search returns them all.

For this particular case, the author maintains a separate CHANGELOG file; however it is harder to find; it would be much better if I could supply the package name and github would return only exact matches (with an option to search only in title, to eliminate hits where a package is mentioned in the list of fixes of another package?)

**Reply by albgus** [reply]
@MylesBorins Not going to sugarcoat it, searching releases is completely broken. If I search for e.g. "1.1.0", the search feature returns everything because it matches every single instance of "1" in the releases. So in practive it's not really possible to search for a specific release version because it's going to match every single release anyway.

Combining that with that preview releases are shown by default finding a specific release usually comes down to going through multiple pages and looking for it manually. I hate it when software projects use the Github releases feature to document their releases, especially when that's the only place where release notes are published.
